integration/Jenkinsfiles/ftest.groovy

```diff
@@ -0,0 +1,177 @@
+                                            """
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                    }
+                },
+                "webdriver" : {
+                    node('SLAVE') {
+                        stage('webdriver') {
+                            ws("$WORKSPACE-webdriver") {
+                                unstash "clone"
+                                timeout(time: 2, unit: 'HOURS') {
+                                    withBuildStatus("$DBPROFILE-$DBVERSION/ftest/cmis", sha) {
+                                        withDockerCompose("$JOB_NAME-$BUILD_NUMBER-webdriver", "integration/Jenkinsfiles/docker-compose-$DBPROFILE-${DBVERSION}.yml", "mvn -B -f $WORKSPACE/nuxeo-distribution/nuxeo-jsf-ui-webdriver-tests/pom.xml clean verify -Pqa,tomcat,$DBPROFILE") {
+                                            archive 'nuxeo-distribution/nuxeo-jsf-ui-webdriver-tests/target/**/failsafe-reports/*, nuxeo-distribution/nuxeo-jsf-ui-webdriver-tests/target/*.png, nuxeo-distribution/nuxeo-jsf-ui-webdriver-tests/target/*.json, nuxeo-distribution/nuxeo-jsf-ui-webdriver-tests/target/**/*.log, nuxeo-distribution/nuxeo-jsf-ui-webdriver-tests/target/**/log/*, nuxeo-distribution/nuxeo-jsf-ui-webdriver-tests/target/**/nxserver/config/distribution.properties, nuxeo-distribution/nuxeo-server-cmis-tests/target/nxtools-reports/*, nuxeo-distribution/nuxeo-jsf-ui-webdriver-tests/target/results/*/*'
+                                            junit '**/target/surefire-reports/*.xml, **/target/failsafe-reports/*.xml, **/target/failsafe-reports/**/*.xml'
+                                            sh """#!/bin/bash -ex
+                                                  ! grep -E '^[0-9]{4}-[0-9]{2}-[0-9]{2}.*ERROR.*' nuxeo-distribution/nuxeo-jsf-ui-webdriver-tests/target/tomcat/log/server.log
+                                            """
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            )
+        }
+    }
+}
+
+// should be in shared library
+
+def warningsPublisher() {
+    step([
+            $class: 'WarningsPublisher',
+	    consoleParsers: [
+                [parserName: 'Maven'],
+                [parserName: 'Java Compiler (javac)']
+	    ]
+	])
+}
+
+def claimPublisher() {
+    step([$class: 'ClaimPublisher'])
+}
+
+def withBuildStatus(String context, String sha, Closure body) {
+    currentBuild.result = 'SUCCESS'
+    setBuildStatus("", "PENDING", context, sha)
+    try {
+        body.call()
+        setBuildStatus("", "SUCCESS", context, sha)
+    } catch (Throwable cause) {
+        if (currentBuild.getResult == null) {
```

Comment:
This check may be redondant with the Jenkins embedded check in the setResult method:
```java
    public void setResult(@Nonnull Result r) {
        if (this.state != Run.State.BUILDING) {
            throw new IllegalStateException("cannot change build result while in " + this.state);
        } else {
            if (this.result == null || r.isWorseThan(this.result)) {
                this.result = r;
                LOGGER.log(Level.FINE, this + " in " + this.getRootDir() + ": result is set to " + r, LOGGER.isLoggable(Level.FINER) ? new Exception() : null);
            }

        }
    }
`

Reply:
Yes this need a cleanup. That's was as try for troubleshooting errors in claim step execution.